apps/prairielearn/src/middlewares/authzCourseOrInstance.ts

```diff
@@ -40,47 +51,321 @@ function clearOverrideCookies(res: Response, overrides: Override[]) {
   });
 }
 
+function verifyAuthnAuthResult(
+  authnAuthResult: CalculateAuthDataSuccessResult['authResult'],
+  viewType: 'student' | 'instructor' | 'none',
+) {
+  // Cannot request a user data override without instructor permissions
+  if (
+    !authnAuthResult.has_course_permission_preview &&
+    !authnAuthResult.has_course_instance_permission_view
+  ) {
+    debug('requested overrides, but authn user does not have instructor permissions');
```

Comment:
Nit, did you find the `debug` prints to be helpful? I'm wondering if we can remove them.

Reply:
I just console log if I need info.

Reply:
Removed in f2f21cbdc2d9d9db149ef73374fb819c857ba348. I think the simplified control flow should make this less needed.